dev-tools/build_release.py

```diff
@@ -618,6 +618,47 @@ def check_norelease(path='src'):
           if pattern.search(line):
             raise RuntimeError('Found //norelease comment in %s line %s' % (full_path, line_number))
 
+def run_and_print(text, run_function):
+  try:
+    print(text, end='')
+    run_function()
+    print(OKGREEN + 'OK' + ENDC)
+  except RuntimeError:
+    print(FAIL + 'NOT OK' + ENDC)
+
+def check_env_var(text, env_var):
+  try:
+    print(text, end='')
+    env[env_var]
+    print(OKGREEN + 'OK' + ENDC)
+  except KeyError:
+    print(FAIL + 'NOT OK' + ENDC)
+
+def check_environment_and_commandline_tools():
+  check_env_var('Checking for AWS env configuration AWS_SECRET_ACCESS_KEY_ID...     ', 'AWS_SECRET_ACCESS_KEY')
+  check_env_var('Checking for AWS env configuration AWS_ACCESS_KEY_ID...            ', 'AWS_ACCESS_KEY_ID')
+  check_env_var('Checking for SONATYPE env configuration SONATYPE_USERNAME...       ', 'SONATYPE_USERNAME')
+  check_env_var('Checking for SONATYPE env configuration SONATYPE_PASSWORD...       ', 'SONATYPE_PASSWORD')
+  check_env_var('Checking for GPG env configuration GPG_KEY_ID...                   ', 'GPG_KEY_ID')
+  check_env_var('Checking for GPG env configuration GPG_PASSPHRASE...               ', 'GPG_PASSPHRASE')
+  check_env_var('Checking for S3 repo upload env configuration S3_BUCKET_SYNC_TO... ', 'S3_BUCKET_SYNC_TO')
+  check_env_var('Checking for git env configuration GIT_AUTHOR_NAME...              ', 'GIT_AUTHOR_NAME')
+  check_env_var('Checking for git env configuration GIT_AUTHOR_EMAIL...             ', 'GIT_AUTHOR_EMAIL')
+
+  run_and_print('Checking command: gpg...            ', partial(check_command_exists, 'gpg', 'gpg --version'))
+  run_and_print('Checking command: expect...         ', partial(check_command_exists, 'expect', 'expect -v'))
+  run_and_print('Checking command: createrepo...     ', partial(check_command_exists, 'createrepo', 'createrepo --version'))
+  run_and_print('Checking command: s3cmd...          ', partial(check_command_exists, 's3cmd', 's3cmd --version'))
+  run_and_print('Checking command: apt-ftparchive... ', partial(check_command_exists, 'apt-ftparchive', 'apt-ftparchive --version'))
+
+  # boto, check error code being returned
+  location = os.path.dirname(os.path.realpath(__file__))
+  command = 'python %s/upload-s3.py' % location
+  run_and_print('Testing boto python dependency...   ', partial(check_command_exists, 'python-boto', command))
```

Comment:
I was not able to make this one becoming GREEN on my laptop though it works when I release plugins.

Basically, running manually:

```sh
python /path/to/elasticsearch/dev-tools/upload-s3.py
```

gives:

```
usage: upload-s3.py [-h] --file path to file [--bucket B42]
                    [--path elasticsearch/elasticsearch] [--key key]
upload-s3.py: error: argument --file/-f is required
```

Do you have a different output?